### ISSUE TITLE
Monitoring improvements for CAPO for GA

### ISSUE THREAD

**Issue by erkanerol** [giantswarm/roadmap#1030]
## User Story

- As a giant swarm customer on OpenStack I want to have a fully working monitoring system.

## Details, Background

We implemented monitoring for CAPO with this epic https://github.com/giantswarm/roadmap/issues/640 but there are some missing elements. We need to complete them for GA. 

- [ ] Monitoring vault (Partially reverting https://github.com/giantswarm/prometheus-rules/pull/314)
- [ ] ~Bastion monitoring~ (not needed for now as we decided to toggle bastions on/off on demand)
- [ ] Do we need to monitor certificate expiration  / automatic rolling?
   - api server and etcd certificate is only rolled during an update 
     - following upstream issue might also very helpful: https://github.com/kubernetes-sigs/cluster-api/issues/6529#issuecomment-1182963211
- [ ] https://github.com/giantswarm/roadmap/issues/1002)
- [x] https://github.com/giantswarm/giantswarm/issues/21653
- [x] https://github.com/giantswarm/giantswarm/issues/22875
- [x] https://github.com/giantswarm/giantswarm/issues/22634
- [ ] https://github.com/giantswarm/roadmap/issues/1209
- [ ] review existing alerting rules before beeing GA
      - based on existing metrics, are thresholds valid and do we have an opsrecipe for paging alerts
      - which metrics are we missing
      - review alerts needed to ask customer for higher ressource quotas

## Blocked by / depends on
-
